sound/soc/sof/intel/hda-dai.c

```diff
@@ -475,36 +475,43 @@ static int ipc4_hda_dai_trigger(struct snd_pcm_substream *substream,
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		snd_hdac_ext_stream_start(hext_stream);
-		if (pipeline->state != SOF_IPC4_PIPE_PAUSED) {
+		if (!pipeline->use_chain_dma) {
+			if (pipeline->state != SOF_IPC4_PIPE_PAUSED) {
+				ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
+								  SOF_IPC4_PIPE_PAUSED);
+				if (ret < 0)
+					return ret;
+				pipeline->state = SOF_IPC4_PIPE_PAUSED;
+			}
+
 			ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
-							  SOF_IPC4_PIPE_PAUSED);
+							  SOF_IPC4_PIPE_RUNNING);
 			if (ret < 0)
 				return ret;
-			pipeline->state = SOF_IPC4_PIPE_PAUSED;
 		}
 
-		ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
-						  SOF_IPC4_PIPE_RUNNING);
-		if (ret < 0)
-			return ret;
 		pipeline->state = SOF_IPC4_PIPE_RUNNING;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_STOP:
 	{
-		ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
-						  SOF_IPC4_PIPE_PAUSED);
-		if (ret < 0)
-			return ret;
+		if (!pipeline->use_chain_dma) {
+			ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
+							  SOF_IPC4_PIPE_PAUSED);
+			if (ret < 0)
+				return ret;
+		}
```

Comment:
@ranj063 this is the sort of things that should be abstracted, this will also be the same in DSP-less mode.